src/core/include/ngraph/specialize_function.hpp

```diff
@@ -4,6 +4,8 @@
 
 #pragma once
 
+#warning("The nGraph API is deprecated and will be removed in the 2024.0 release. For instructions on transitioning to the new API, please refer to https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html")
+
 #include "ngraph/function.hpp"
 
 namespace ngraph {
```

Comment:
and we can deprecated even namespace https://stackoverflow.com/questions/22806878/is-there-a-way-to-deprecate-a-namespace